projects/soft_999_eric_zhao_fw_103/src/main.c

```diff
@@ -0,0 +1,57 @@
+#include "adc.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+void prv_analog_read_a6(const GpioAddress *address, void *context) {
+  uint16_t a6_data = 0;
+  GpioAddress *a6_addr = context;
+
+  if (adc_read_raw_pin(*a6_addr, &a6_data) == STATUS_CODE_OK) {
+    LOG_DEBUG("A6 data: %d\n", a6_data);
+  } else {
+    LOG_DEBUG("Failed to read A6 data");
+  }
+}
+
+int main(void) {
+  GpioAddress a6_reading_addr = {
```

Comment:
An alternate way of doing these gpio declarations that we use frequently to reduce clutter is
`#define ADC_ADDRESS {.port = GPIO_PORT_A, .pin = 6}`
This allows us to in main.c simply just do the assignment as 
`GpioAddress adc_ad = ADC_ADDRESS` which will insert the struct we defined above. 

The way you've done it is fine, just a thought :)